### PR TITLE
Improved touch event handling for drawing tools

### PR DESCRIPTION
**Pull request by eatyourgreens**
- style the SVG subject with [`touch-action: pinch-zoom`](https://developer.mozilla.org/en-US/docs/Web/CSS/touch-action).
- cancel both the default handler and event bubbling for all pointer events on the drawing canvas.

This PR should apply the same changes to the drawing tools that we added to PFE in https://github.com/zooniverse/Panoptes-Front-End/pull/5411.

## Package
lib-classifier

## Linked Issue and/or Talk Post
- Towards #3534

## How to Review
Marsh Explorer is a good project to check the drawing tools on touch devices. Basically, we expect that you should be able to draw without triggering any default browser behaviours, such as scrolling the screen when you drag to create a mark.
https://www.zooniverse.org/projects/marshexplorer/marsh-explorer/classify

In the dev classifier: https://local.zooniverse.org:8080/?project=marshexplorer/marsh-explorer&env=production

This branch:
https://fe-project-branch.preview.zooniverse.org/projects/marshexplorer/marsh-explorer/

# Checklist
_PR Creator - Please cater the checklist to fit the review needed for your code changes._
_PR Reviewer - Use the checklist during your review. Each point should be checkmarked or discussed before PR approval._

## General
- [x] Tests are passing locally and on Github
- [ ] Documentation is up to date and changelog has been updated if appropriate
- [x] You can `yarn panic && yarn bootstrap` or `docker-compose up --build` and FEM works as expected
- [ ] FEM works in all major desktop browsers: Firefox, Chrome, Edge, Safari (Use Browserstack account as needed)
- [ ] FEM works in a mobile browser

## General UX
Example Staging Project: [i-fancy-cats](https://local.zooniverse.org:3000/projects/brooke/i-fancy-cats)
- [ ] All pages of a FEM project load: Home Page, Classify Page, and About Pages
- [ ] Can submit a classification
- [ ] Can sign-in and sign-out
- [ ] The component is accessible
  - Can be used with a screen reader [BBC guide to VoiceOver](https://bbc.github.io/accessibility-news-and-you/assistive-technology/testing-steps/voiceover-mac.html)
  - Can be used from the keyboard [WebAIM guide to keyboard testing](https://webaim.org/techniques/keyboard/#testing)
  - It is passing accessibility checks in the storybook

## Bug Fix
- [x] The PR creator has listed user actions to use when testing if bug is fixed
- [ ] The bug is fixed
- [x] Unit tests are added or updated

### CODE REVIEW

**Review by shaunanoordin — COMMENTED** [review]
The code changes here are very straightforward and make sense, but unfortunately I'm running a bit late to get this reviewed for Maintenance Week. @mcbouslog can I please pass the review baton to you?

Testing notes:
- Basic functionality check:
  - Volunteer should be able _click_ to place a Point.
  - Volunteer should be able to _drag_ to move a Point.
  - Volunteer should be able to _click_ another Point to select it
  - Volunteer should be able to _drag_ to place a Rectangle. (Or Circle, if you prever)
  - Volunteer should be able to _drag_ to (1) resize and (2) move a Rectangle.
  - Volunteer should be able to do all the above without accidentally scrolling the webpage view.
- This needs to be tested across multiple browsers (Chrome/Safari/Firefox) on desktop and touchscreen devices (Chrome emulation/Browserstack/use your actual phone) _(Sorry, this is the one that's a big time-eater 😢 )_
- You can use my Drawing WF, since it has a lot of tools set up: https://frontend.preview.zooniverse.org/projects/darkeshard/test-project-2022/classify/workflow/3596?env=staging

⚠️ Note 1: I'm not entirely sure if this PR can 100% solve all touch interactions on mobile, so if in doubt, please compare it to PFE to see if it at least brings the functionality on par with our old website: https://master.pfe-preview.zooniverse.org/projects/darkeshard/test-project-2022/?env=staging (choose "Drawing WF")

Note 2: for context, the **Point tool** is probably the most important to check for, since ]PenguinWatch](https://www.zooniverse.org/projects/penguintom79/penguin-watch) is one of the major projects we want to move over to FEM, and we want it to continue working on mobile & tablets.

If you don't have time, please reassign back to me and I'll take care of this next week. Thanks, Mark!

**Comment by eatyourgreens** [comment]
Note that transcription lines crash the classifier on touch screens. That bug is fixed by #3536.

**Review by mcbouslog — COMMENTED** [review]
The code LGTM, but noted a few issues below when testing on https://fe-project-branch.preview.zooniverse.org/projects/darkeshard/test-project-2022/classify/workflow/3596?env=staging&demo=true .

I focused on testing using a few personal devices due to Browserstack being frustrating to use for reasons unrelated to this PR, with long delays in drawing actions.

Worked great
- test point, rectangle, freehand on Chrome desktop
- test point, rectangle, freehand on Firefox desktop
- test point, rectangle, freehand on Safari desktop

Issues 
- on the iPad, iOS 15.6.1, Chrome, Safari, and Firefox
    - if I touched then dragged (not quickly, but without hesitating) the rectangle or freehand tool wouldn't draw the intended shape and would scroll unexpectedly
    - if I touched, paused, then dragged the subject viewer would get the blue focus/selected indicator and it would draw the shape I wanted, but then trying to draw another mark the same way (touch, pause, drag) would cause an application error
- on Pixel 3, Android 12
  - Chrome - can't draw rectangle/freehand, can't move point
  - Firefox - with quick touch and drag scrolls and can't draw intended shape, with touch, pause, then draw only goes left/right
  
I can upload videos or screenshots if it would be helpful, or I might be doing something wrong on my end, if @eatyourgreens maybe you could check https://fe-project-branch.preview.zooniverse.org/projects/darkeshard/test-project-2022/classify/workflow/3596?env=staging&demo=true on your iPhone or if you have an iPad and confirm just one of issues noted to see if something on my end or confirm?

**Comment by eatyourgreens** [comment]
On iOS 15, I can confirm that Marsh Explorer works in PFE but not on this branch. 🤔

**Comment by eatyourgreens** [comment]
Also noting that I can pan and zoom on PFE but not the new classifier. That will be a big problem for Penguin Watch.

**Comment by eatyourgreens** [comment]
I'm deploying this again, but with `touch-action: pinch-zoom` applied to the `svg` element.

**Comment by eatyourgreens** [comment]
Not related to this PR, but there’s a bug with the position of the delete button for circles. 

Screenshot showing the delete button positioned very far from a drawn circle on a mobile device.

**Comment by eatyourgreens** [comment]
The polygon tool works on mobile too, on this branch, without triggering #3498. That's interesting. I think #3536 might still be needed, to make the code more robust against invalid reference errors.

**Comment by eatyourgreens** [comment]
Freehand lines also work on this branch, in mobile Safari.

**Comment by eatyourgreens** [comment]
TBH, pinch and zoom on iOS is a lot easier to use, with the drawing tools, than the built-in zoom buttons.

**Comment by eatyourgreens** [comment]
I'm going to deploy this again, now that it includes #3536.

**Comment by eatyourgreens** [comment]
Marsh Explorers still works on my phone, after the latest rebase and deploy. Needs checking on Android.

**Comment by eatyourgreens** [comment]
Rebasing this and deploying again.

**Comment by eatyourgreens** [comment]
Noting that this PR blocks #3571. We'd like to test that out on `fe-branch` once this branch is merged.

**Review by mcbouslog — APPROVED** [review]
This is working much better! I confirmed:

iOS
- Chrome, Firefox, and Safari - I can draw/move/delete point, rectangle, and freehand line

Android
- Chrome - I can draw/move/delete point, rectangle, and freehand line

I think I'm seeing issues with Android Firefox where shapes can only be drawn horizontally, and I think there's an error with the freehand line tool if I press undo many times, but I don't think either of those issues should hold this up.

👍  :shipit:

**Pull request merged**
